dipy/tracking/tests/test_tracking.py

```diff
@@ -956,3 +956,30 @@ def test_affine_transformations():
         npt.assert_(np.allclose(streamlines_inv[0], expected[0], atol=0.3))
         npt.assert_equal(len(streamlines_inv[1]), len(expected[1]))
         npt.assert_(np.allclose(streamlines_inv[1], expected[1], atol=0.3))
+
+
+def test_random_seed_initialization():
+    """Test that the random generator can be initialized correctly with the
+    tracking seeds.
+    """
+    sphere = HemiSphere.from_sphere(unit_octahedron)
+    pmf = np.zeros((4, 4, 4, 3))
+    x = np.array([0., 0, 0, 57.421434502602544])
+    y = np.array([0., 1, 2, 21.566539227085478])
+    z = np.array([1., 1, 1, 51.67881720942744])
+
+    seeds = np.row_stack([np.column_stack([x, y, z]),
+                          np.random.random((10, 3))])
+    sc = BinaryStoppingCriterion(np.ones((4, 4, 4)))
+    dg = ProbabilisticDirectionGetter.from_pmf(pmf, 60, sphere)
+
+    randoms_seeds = [None, 0, 1, -1] + list(np.random.random(10)) \
+        + list(np.random.randint(0, np.iinfo(np.int32).max, 10))
+
+    for rdm_seed in randoms_seeds:
+        _ = Streamlines(LocalTracking(direction_getter=dg,
+                                      stopping_criterion=sc,
+                                      seeds=seeds,
+                                      affine=np.eye(4),
+                                      step_size=1.,
+                                      random_seed=rdm_seed))
```

Comment:
I would add a small test to make sure it fails if you put a `random seed  = np.iinfo(np.int32).max + 1`.

Actually, What is error when this happens (I know, it should not)

Reply:
`random_seed` is not directly input to the random generator. It is a hash of `random_seed` and the tractography seed position:

```
s_random_seed = hash(np.abs((np.sum(s)) + self.random_seed)) \
                    % (2**32 - 1)
random.seed(s_random_seed)
np.random.seed(s_random_seed)
fast_numpy.seed(s_random_seed)
```

It is a `OverflowError`. I added a test in `fast_numpy`.

Reply:
What I mean is:  if I do naively that:

```python
randoms_seeds = np.iinfo(np.uint32).max + 1    #. or  -1
_ = Streamlines(LocalTracking(direction_getter=dg,
                                      stopping_criterion=sc,
                                      seeds=seeds,
                                      affine=np.eye(4),
                                      step_size=1.,
                                      random_seed=randoms_seeds))
```

I get this error:

```python
Traceback (most recent call last):
  File "/Users/skoudoro/devel/dipy/dipy/tracking/tests/test_tracking.py", line 988, in <module>
    test_random_seed_initialization()
  File "/Users/skoudoro/devel/dipy/dipy/tracking/tests/test_tracking.py", line 981, in test_random_seed_initialization
    _ = Streamlines(LocalTracking(direction_getter=dg,
  File "/opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/lib/python3.9/site-packages/nibabel/streamlines/array_sequence.py", line 148, in __init__
    self.extend(iterable)
  File "/opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/lib/python3.9/site-packages/nibabel/streamlines/array_sequence.py", line 324, in extend
    for e in elements:
  File "/opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/lib/python3.9/site-packages/dipy/tracking/utils.py", line 886, in transform_tracking_output
    for sl in streamlines:
  File "/opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/lib/python3.9/site-packages/dipy/tracking/local_tracking.py", line 140, in _generate_tractogram
    fast_numpy.seed(s_random_seed)
  File "fast_numpy.pyx", line 188, in dipy.utils.fast_numpy.seed
OverflowError: value too large to convert to int
```

I feel it is easy to prevent, via a small check in the `LocalTracking  __init__`

Reply:
not all users will understand this `OverflowError: value too large to convert to int`